libs/input-mapping/src/Staging/Definition.php

```diff
@@ -1,178 +1,58 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Keboola\InputMapping\Staging;
 
-use Keboola\InputMapping\Exception\StagingException;
+use Keboola\InputMapping\File\StrategyInterface as FileStrategyInterface;
+use Keboola\InputMapping\Table\StrategyInterface as TableStrategyInterface;
 
-class Definition
+class Definition extends AbstractDefinition
```

Comment:
Kdyz si cast toho vytahl do `AbstractDefinition`, predpokladam, ze dal (v OM?) bude jina definition, ktera z ni bude dedit. Tim padem bych `Definition` prejmenoval na neco konkretnejsiho, protoze to uz neni proste genericka `Definition`, ale je pro nejakyt konkretni case.

Reply:
je to `Keboola\InputMapping\Staging\Definition`, OM má `Keboola\OutputMapping\Staging\Definition` https://github.com/keboola/platform-libraries/blob/odin-PST-601/libs/output-mapping/src/Staging/Definition.php

Nečistý je to v tom, že 
`Keboola\OutputMapping\Staging\Definition extends Keboola\InputMapping\Staging\AbstractDefinition`, ale už to není naštěstí 
`Keboola\OutputMapping\Staging\Definition extends Keboola\InputMapping\Staging\Definition` - takže to by se ještě hodilo vytáhnout tu abstract class někde bokem, pířpadně i s interfacem

Reply:
No spis jsem narazel na to, ze kdyz nekde v kodu uvidis `new Definition`, vis kulovy, co to je - `definition` samo o sobe je hrozne obecny pojmenovani. Takze bych mirik k necemu jako `AbstractStagingDefinition` + `InputStagingDefinition`. Stejny problem mam s `Client` class, kterou si vsude aliasuju :slightly_smiling_face:

Reply:
Aha, jasně, chápu. No tak to bez dovolení řadím taky do kategorie "určitě, ale ne tady a ne teď " :) Takovejch tříd je tam víc Strategy, StrategyFactory